### PR TITLE
[PP] Support extra loss_fn kwargs in pipeline schedules

### PR DESCRIPTION
**Pull request by wwwjn**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #181057

Add a `loss_kwargs` parameter threaded through `step()` →
`_step_microbatches()` → `_maybe_compute_loss()` → `_compute_loss()`
and through `_initialize_stage()` → `_initialize_pp_stages()` →
`_prepare_backward_infra()` → `_backward_metadata_inference()`.

This enables loss functions that require additional arguments beyond
`(output, target)`, such as chunked cross-entropy loss which needs
`global_valid_tokens` for proper scaling:

    schedule.step(..., loss_kwargs={"global_valid_tokens": tokens})

The kwargs are passed as a parameter rather than stored on the schedule
object, since they vary across steps and should not persist as state.

Without this, `_backward_metadata_inference` and `_compute_loss` call
`loss_fn(output, target)` with only 2 args, causing a TypeError when
the loss function requires additional keyword arguments.

### CODE REVIEW

**Review by tianyu-l — COMMENTED** [review]
looks reasonable to me, I'll let @sanketpurandare stamp

**Review by sanketpurandare — APPROVED** [review]
Thanks for the change. Just fix the lint please
